src/pages/basic_data/institution_admin/edit_user/UserFormDialog.tsx

```diff
@@ -0,0 +1,161 @@
+  onClose: () => void;
+}
+
+export interface UserFormData {
+  person?: CristinPerson;
+  user?: InstitutionUser;
+}
+
+export const UserFormDialog = ({ open, onClose, existingPerson }: UserFormDialogProps) => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const user = useSelector((store: RootState) => store.user);
+  const topOrgCristinId = user?.topOrgCristinId;
+  const customerId = user?.customerId ?? '';
+
+  const personCristinIdentifier = getIdentifierFromId(existingPerson.id);
+  const topOrgCristinIdentifier = topOrgCristinId ? getIdentifierFromId(topOrgCristinId) : '';
+  const username =
+    personCristinIdentifier && topOrgCristinIdentifier ? `${personCristinIdentifier}@${topOrgCristinIdentifier}` : '';
+
+  const institutionUserQuery = useQuery({
+    enabled: open && !!username,
+    queryKey: [username],
+    queryFn: () => fetchUser(username),
+    meta: { errorMessage: false }, // No error message, since a Cristin Person will lack User if they have not logged in yet
+    retry: false,
+  });
+
+  const personMutation = useMutation({
+    mutationFn: async (person: CristinPerson) => {
+      if (!person.verified) {
+        person.keywords = undefined;
+      }
+
+      return await updateCristinPerson(person.id, person);
+    },
+    onError: () => dispatch(setNotification({ message: t('feedback.error.update_person'), variant: 'error' })),
+  });
+
+  const userMutation = useMutation({
+    mutationFn: async (user: InstitutionUser) => {
+      const filteredRoles = !user.roles.some((role) => role.rolename === RoleName.PublishingCurator)
+        ? user.roles.filter(
+            (role) => role.rolename !== RoleName.CuratorThesis && role.rolename !== RoleName.CuratorThesisEmbargo
+          )
+        : user.roles;
+      user.roles = filteredRoles;
+      if (institutionUserQuery.isSuccess) {
+        return await updateUser(user.username, user);
+      } else {
+        return await createUser({
+          customerId,
+          roles: user.roles,
+          nationalIdentityNumber: getValueByKey('NationalIdentificationNumber', existingPerson.identifiers),
+        });
+      }
+    },
+    onError: () =>
+      dispatch(setNotification({ message: t('feedback.error.update_institution_user'), variant: 'error' })),
+    onSuccess: () =>
+      dispatch(setNotification({ message: t('feedback.success.update_institution_user'), variant: 'success' })),
+  });
+
+  const initialValues: UserFormData = {
+    person: existingPerson,
+    user: institutionUserQuery.isError
+      ? {
+          institution: customerId ?? '',
+          roles: [{ type: 'Role', rolename: RoleName.Creator }],
+          username: username,
+        }
+      : institutionUserQuery.data,
+  };
+
+  return (
+    <Dialog open={open} onClose={onClose} maxWidth="xl" fullWidth transitionDuration={{ exit: 0 }}>
+      <DialogTitle>{t('basic_data.person_register.edit_person')}</DialogTitle>
+
+      <Formik
```

Comment:
Dette kunne gjerne vært delt opp i to komponenter, da det vil være behov for å gjøre dette i både dialog og inline senere

Reply:
neh, la oss løse problemet når det oppstår :)